### PR TITLE
Merge asplos into main

### PR DESCRIPTION
**Pull request by AndraBisca**

### CODE REVIEW

File: programming_examples/lit.cfg.py [hunk 1]
```diff
@@ -19,9 +21,12 @@
 # Configuration file for the 'lit' test runner.
 
 # name: The name of this test suite.
-config.name = "AIE_REFERENCE_DESIGNS"
+config.name = "AIE_TUTORIALS"
 
 config.test_format = lit.formats.ShTest(not llvm_config.use_lit_shell)
+config.environment["PYTHONPATH"] = "{}".format(
+    os.path.join(config.aie_obj_root, "python")
+)
```
Comment (fifield) [inline programming_examples/lit.cfg.py:29]: handled by line 51?

File: programming_examples/lit.cfg.py [hunk 2]
```diff
@@ -7,9 +7,11 @@
 # Copyright (C) 2022, Advanced Micro Devices, Inc.
 
 import os
+import platform
```
Comment (fifield) [inline programming_examples/lit.cfg.py:10]: unused

File: programming_examples/lit.cfg.py [hunk 3]
```diff
@@ -7,9 +7,11 @@
...
 import re
 import shutil
 import subprocess
+import tempfile
```
Comment (fifield) [inline programming_examples/lit.cfg.py:14]: unused

File: aie_kernels/aie2/conv2dk1_i8.cc
Comment (fifield) [inline aie_kernels/aie2/conv2dk1_i8.cc:1]: this undoes clang-format?

File: aie_kernels/aie2/conv2dk1_i8.h
Comment (fifield) [inline aie_kernels/aie2/conv2dk1_i8.h:1]: just a mode change, remove

File: aie_kernels/aie2/conv2dk1_i8.cc
Comment (AndraBisca) [reply aie_kernels/aie2/conv2dk1_i8.cc:1]: I think the file I kept from asplos didn't have the formatting. I just applied the patch, so it should be good now?

File: programming_examples/basic/vector_reduce_min/run_vck5000.lit [hunk 4]
```diff
@@ -0,0 +1,9 @@
+// (c) Copyright 2023 Advanced Micro Devices, Inc.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+// REQUIRES: hsa
+//
+// RUN: %python %S/aie2.py xcvc1902 6 > ./aie.mlir
+// RUN: %python aiecc.py %VitisSysrootFlag% --host-target=%aieHostTargetTriplet% %link_against_hsa% ./aie.mlir -I%host_runtime_lib%/test_lib/include %extraAieCcFlags% %S/test_vck5000.cpp -o test.elf -L%host_runtime_lib%/test_lib/lib -ltest_lib
+// RUN: %run_on_vck5000 ./test.elf
+
```
Comment (fifield) [inline programming_examples/basic/vector_reduce_min/run_vck5000.lit:9]: remove ?

Comment (AndraBisca) [reply programming_examples/basic/vector_reduce_min/run_vck5000.lit:9]: Will do!

**Review by fifield — APPROVED** [review]

**Review by jgmelber — APPROVED** [review]

**Review by denolf — APPROVED** [review]

**Review by jackl-xilinx — APPROVED** [review]

**Pull request merged**